modules/optimization/doc/content/modules/optimization/examples/material_transient.md

```diff
@@ -0,0 +1,89 @@
+by leveraging inverse optimization theory, see [section](theory/InvOptTheory.md#sec:material_inversion). As
+with [material inversion with constant conductivity](materialInv_ConstK.md), this is a nonlinear optimization
+problem that requires an appropriate initial guess and parameter bounds to obtain convergence.
+
+## Transient Problem Definition
+
+The problem analyzed here is a two-dimensional diffusion example with square geometry. The right
+and top boundaries are subjected to zero Dirichlet boundary conditions. The spatial distribution of
+the material parameter $D$ is given in [diff_parameter]. While this material parameter is known in the forward
+problem, its distribution to minimize the solution misfit is optimized in the adjoint pass.
+
+
+!media large_media/optimization/material_transient_diff.png
+       style=width:40%;margin:auto;padding-top:2.5%;background-color: white;color: black;
+       id=diff_parameter
+       caption=Spatial parameterization of the diffusivity parameter for the transient problem.
+
+As this is a transient or non-steady problem, the solution at each time step depends on the solution
+at previous steps. The forward problem is solved as usual in MOOSE, using implicit Euler time stepping.
+The solution to the diffusion problem is obtained for ten steps with a time step of 0.1 seconds. The
+solution to the diffused variable at time $t=0.6$ s, together with the measurement points provided for
+material inversion optimization, is shown in [solution].
+
+!media large_media/optimization/material_transient_solution.png
+       style=width:40%;margin:auto;padding-top:2.5%;background-color: white;color: black;
+       id=solution
+       caption=Snaphot of the forward simulation at the end of the sixth time step (0.6 s). The yellow
+       marking points to the measurement points provided in the procedure for obtaining
+       the optimal diffusivity parameters of the material at the end of the adjoint problem.
+
+The solution to the adjoint problem requires an initial solution vector with estimates of the material
+property values. In addition to that, bounds for those parameter need to be provided. In [sensitivity],
+we plot the objective function as we vary the initial estimate for the diffusion parameter at the location
+(0.75, 0.75). The slope of the objective function with respect to variations in the parameter can have
+implication in the overall convergence of the adjoint problem.
+
+!media large_media/optimization/material_transient_of.png
+       style=width:40%;margin:auto;padding-top:2.5%;background-color: white;color: black;
+       id=sensitivity
+       caption=Sensitivity of the objective function when changing the diffusivity parameter at
+       the location (0.75, 0.75) around its optimal value '0.05'.
+
+
+## Problem Input File Setup
+
+This example makes use of the [TransientAndAdjoint](TransientAndAdjoint.md) executioner, which coordinates
```

Comment:
```suggestion
This example makes use of the [TransientAndAdjoint](TransientAndAdjoint.md) executioner, which automatically computes the adjoint by transposing the forward problem's Jacobian.  This frees the user from having to derive and implement and an adjoint input file based on the forward problem kernels.  This also solves the forward and adjoint problems in a single sub-application.  An alternative method
```

Reply:
I don't understand "derive and implement" an adjoint input file part. Also, the "single sub-application": Aren't we simply running a MOOSE module?

Reply:
Maybe sub-application is the wrong wording.   We are just running a moose sub-app or child-app.  I was trying to say  the old method runs two seperate sub-apps, one for the forward problem and a completely seperate sub-app for the adjoint problem.  With TransientAndAdjoint, we only run a single sub-app that solves the forward and adjoint problem at the same time using the multi-executioner system.